### DISCUSSION TITLE
2024/11/20/tweaking-stunt-island

### DISCUSSION THREAD

**Discussion by giscus** [AlbertoMarnetto/annali-discussions#1] [Announcements]
# 2024/11/20/tweaking-stunt-island

http://localhost:4000/2024/11/20/tweaking-stunt-island.html

**Comment by AlbertoMarnetto** [comment]
### Related discussions

* https://news.ycombinator.com/item?id=42305954
* https://old.reddit.com/r/ReverseEngineering/comments/1h5qln5/tweaking_stunt_islands_30yearold_3d_engine/
* https://old.reddit.com/r/retrogamedev/comments/1h8f89o/800_detail_tweaking_stunt_islands_30yearold_3d/
* https://old.reddit.com/r/dosgaming/comments/1h8f8id/800_detail_tweaking_stunt_islands_30yearold_3d/
* https://hackaday.com/2024/12/05/cranking-up-the-detail-in-a-flight-simulator-from-1992/
* https://korben.info/amelioration-graphique-stunt-island-assembleur.html
* https://www.high-voltage.cz/sortky/letecky-simulator-stunt-island-1992-nyni-s-800-detaily/
* https://www.vogons.org/viewtopic.php?t=104051

**Comment by ShhhhDigitalMedia** [comment]
You've certainly created a gem of detective work into the genius of the coding behind Disney's Stunt Island, despite the fact that its hidden behind the obfuscation of a 1990s C/C++/ASM compiler veil.  Reminds me of the old days. Coding scrolling text demos for VGA 13h 320x200 or 640x480 interleaved video memory.

What a wonderful thing you've created here. Reminds me of the movie version of Ready Player One, and the Easter egg mysteries left behind by the Atari 2600's "Adventure" cartridge coder. 

Have you looked to the Elite community, who have extensive experience in having reverse engineered all three versions of the David Braben/Ian Bell masterpiece from their binaries alone?

I probably won't try the patch, but its great to see that someone else is interested in doing this. Like digital archaeology of a sort, where the crypts we explore nowadays are measured in decades rather than millennia.

I'd love to see Disney pick this gem title up, and produce a sequel, using modern technology and re-hiring as many of the original devs who are still alive. In nobody does this, I might pursue a freeware version with Unreal Engine, or at the very least, propose such an endeavor to the open source community. I wonder how many modern film makers Stunt Island inspired?

**Reply by AlbertoMarnetto** [reply]
Thanks for your comment! I am not in contact with the Elite community but I got into interesting discussions with people working on other old games. And I totally agree with your metaphor of digital archaeology and the thrill of uncovering old secrets, as trivial as they might be.

**Comment by CommanderGrog** [comment]
First of all, thank you very much for this fantastic work. Seriously, this is exactly the kind of deep-dive tinkering that keeps these old gems alive, and I really enjoyed both the write-up and the results.

I do have a small issue, though, and I’m not sure if I’m doing something wrong or if it’s an edge case:

The patch works perfectly fine with the standard, unmodified executable from the GOG version. No problems at all there.

However, if I first apply Disney’s official Patch #3 (which itself works flawlessly), and then run your si-8x.exe, the resulting stunt1.exe no longer starts in DOSBox. I just get a blinking cursor and nothing else happens.

Do you have any idea what might be causing this, or anything I could try on my end? Maybe a mismatch in the compression step or a slightly different binary layout in the patched version?

Either way, thanks again for sharing this. It’s genuinely impressive work.

**Reply by AlbertoMarnetto** [reply]
Hello, @CommanderGrog ! Are you the one of the same-named Youtube channel?

I'm happy that someone finally managed to test the patch on the #<b></b>3 executable! You did everything right, the error was on my side. I had amended the modding program to support Disney's #<b></b>3 update, but in the .7z archive I published I mistakenly put an older version, which only supported the original exe.

Now the error should be fixed. I have published a new patch version: on my PC it produces the correct executable, please try it and let me know if it works on your machine too.

By the way, your comment arrives in a good moment as I am again playing with *Stunt Island*'s internals. But whether something concrete will come out of it, I do not know yet.

Thanks for your message and wish you good retrogaming!

**Reply by CommanderGrog** [reply]
Yes, that’s me :D

Thanks a lot for the quick fix and the explanation, that makes perfect sense. I’ll give the updated patch a try tomorrow after work and will report back how it behaves on my setup.

Really appreciate the effort you’re putting into this. It’s fascinating to see Stunt Island being explored and improved like this, all these years later.

Looking forward to seeing if anything more comes out of your current experiments 🙂

Thanks again!

edit: And no surprises there, it works perfectly witch Patch #3 now. Always loved this game, thank you for making it a little better!

(I just found your Patch for Stunts/4D Sports Driving. Can't wait to try this out soon. I've played Stunts for ages and shared so many cruel tracks with my classmates.)

**Reply by AlbertoMarnetto** [reply]
Happy to hear it is now working. Good flights, and take care! And if you pick up *Stunts*, consider visiting [the community site](https://stunts.hu/), at least to check out some of the cars and mods that are continuously being created!
